### ISSUE TITLE
Cover/Front-Matter aligns incorrectly to odd/even pages

### ISSUE THREAD

**Issue by bublath** [ChordPro/chordpro#251]
When adding a cover, an empty page is added if the cover has 2 pages - for even-odd-pages=1, I would expect this the other way round, otherwise the whole alignment with page numbers on the outside gets mixed up with normal double sided printing.

I modified the condition on PDF.pm around line 250 to
	  if $ps->{'even-odd-pages'} && !($page % 2);
(adding a "1") and it now looks ok for me.

I'm not sure if even-odd-pages=-1 needs some special handling, as I don't understand the use case.

**Comment by sciurius** [comment]
Can you give a context diff (or, better still) a diff w.r.t. the dev branch?
This particular line of code occurs at several places.

**Comment by sciurius** [comment]
5.988_039 adds some testing materials for page arrangements.

**Comment by bublath** [comment]
This commit to my development should make it clear (as it is against a pretty recent dev branch)
https://github.com/bublath/chordpro/commit/dc31ecbe5ceda543ea981d61e56ec696d34025ae?diff=unified
It in addition contains an extension, so "front-matter" is available as config option.
My goal is to always just do a "chordpro *.chopro -o output.pdf" per directory without having to worry about any commandline settings. So being able to set the "front-matter" (probably should also add "cover" as alias), seems a natural extension.

In my approach the command-line would still override the config, which seems to me to most intuitive as it is the most visible place to the user. I think I noticed that you did this the other way round for some option. Maybe something to consider.

When I run your new test with my patch, it fails as my output looks like this, which I believe is right (I also preserved the pdf and checked it with a PDF viewer). Only thing that puzzles me is that the second toc on page 5 is not listed here.
```
title;pages;sort title;artists;composers;collections;keys;years
__front_matter__;1;Front Matter;ChordPro;;;;
__table_of_contents__;3-4;Table of Contents;ChordPro;;;;
Song1;7;;Artist1;Composer1;;A;
Song2;9;;Artist2;Composer2;;B;
Song3;11;;Artist3;Composer3;;C;
__back_matter__;12;Back Matter;ChordPro;;;;
```

**Comment by sciurius** [comment]
Good idea to add `front-matter` and `back-matter` to the config. By default, command line options should always override config settings, i.e. no special tricks needed if you copy them in Config.pm. See the attached patch.
Where did you notice the 'other way round'?

[config-matter.patch.txt](https://github.com/ChordPro/chordpro/files/9641132/config-matter.patch.txt)

The page alignment fix (after front-matter) is ok, that was wrong in the current code.

> Only thing that puzzles me is that the second toc on page 5 is not listed here.

The toc pages counting (and reporting in the CSV) is not really dealing well with multiple TOCs. The page count that you see is the pages of toc1 (1) + toc2 (1) = 2. The most correct solution would be to treat the TOCs individually, 
e.g.
````
title;pages;sort title;artists;composers;collections;keys;years
__front_matter__;1;Front Matter;ChordPro;;;;
__table_of_contents__;3;Table of Contents;ChordPro;;;;
__table_of_contents_2__;5;Contents by Title;ChordPro;;;;
Song1;7;;Artist1;Composer1;;A;
Song2;9;;Artist2;Composer2;;B;
Song3;11;;Artist3;Composer3;;C;
__back_matter__;12;Back Matter;ChordPro;;;;
````

**Comment by bublath** [comment]
I almost assumed that there would be a more simple way - well I'm still learning about the code :)
On the "other way round": I just remember that I had doubts about my implemention as I believed I saw a setting that was overriding the command line. I did not make a note and fail to find it again. Don't bother, probably misread it.
On toc: Above looks good and consistent.

**Comment by sciurius** [comment]
I currently have no plans to enhance the toc entries in the csv. Can you make an issue so that it will not be forgotten?

**Comment by sciurius** [comment]
BTW, FYI: The entries in the schema file are (or should be) alphabeticallly ordered. For no particular reason ☺.

**Comment by sciurius** [comment]
Config entries for front-matter and back-matter are in 5.988_040.
Issue 251 is fixed in 5.988_041.
Thanks for the input and the ideas.
